components/scream/src/physics/shoc/shoc_diag_second_moments_lbycond_impl.hpp

```diff
@@ -0,0 +1,50 @@
+
+#ifndef SHOC_DIAG_SECOND_MOMENTS_LBYCOND_IMPL_HPP
+#define SHOC_DIAG_SECOND_MOMENTS_LBYCOND_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+#include "physics_functions.hpp" // also for ETI not on GPUs
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::shoc_diag_second_moments_lbycond(
+    const Scalar& wthl_sfc, const Scalar& wqw_sfc, const Scalar& uw_sfc, const Scalar& vw_sfc,
+    const Scalar& ustar2, const Scalar& wstar,
+    Scalar& wthl_sec, Scalar& wqw_sec, Scalar& uw_sec, Scalar& vw_sec,
+    Scalar& wtke_sec, Scalar& thl_sec, Scalar& qw_sec, Scalar& qwthl_sec)
+{
+  // Purpose of this subroutine is to diagnose the lower
+  // boundary condition for the second order moments needed
+  // for the SHOC parameterization.
+  // The thermodymnamic, tracer, and momentum fluxes are set
+  // to the surface fluxes for the host model, while the
+  // thermodynamic variances and covariances are computed
+  // according to that of Andre et al. 1978.
+
+  const auto ufmin = 0.01;
+
+  auto uf = sqrt(ustar2+sp(0.3)*wstar*wstar);
+  uf = max(ufmin,uf);
+
+  // Diagnose thermodynamics variances and covariances
+  thl_sec   = sp(0.4)*sp(1.8)*(wthl_sfc/uf)*(wthl_sfc/uf);
+  qw_sec    = sp(0.4)*sp(1.8)*(wqw_sfc/uf)*(wqw_sfc/uf); 
```

Comment:
Can someone remind me whether we advocate the use of `square` for expressions like `(wqw_sfc/uf)*(wqw_sfc/uf)` above in C++ (and `cube` for those like on line 46 below)?

Reply:
@jeff-cohere @ambrad There is no square and cube function that take Scalar arguments, that is the reason why it is unrolled in the code.

Reply:
See https://github.com/E3SM-Project/scream/blob/b2349c327fc5f564860d3649b97dfe61a1a42b8f/components/cam/src/physics/cam/bfb_math.inc